django/db/migrations/autodetector.py

```diff
@@ -1187,14 +1192,68 @@ def create_altered_indexes(self):
 
             old_indexes = old_model_state.options[option_name]
             new_indexes = new_model_state.options[option_name]
-            add_idx = [idx for idx in new_indexes if idx not in old_indexes]
-            rem_idx = [idx for idx in old_indexes if idx not in new_indexes]
+
+            added_indexes = [idx for idx in new_indexes if idx not in old_indexes]
+            removed_indexes = [idx for idx in old_indexes if idx not in new_indexes]
+            remove_from_added = []
+            remove_from_removed = []
+            renamed_indexes = []
+            for added_index in added_indexes:
+                for removed_index in removed_indexes:
+                    if (
+                        added_index.equal_except_name(removed_index)
+                        and added_index != removed_index
+                    ):
+                        remove_from_added.append(added_index)
+                        remove_from_removed.append(removed_index)
+                        renamed_indexes.append((added_index, removed_index, None))
+            # Remove renamed indexes from added and removed indexes.
+            added_indexes = [
+                idx for idx in added_indexes if idx not in remove_from_added
+            ]
+            removed_indexes = [
+                idx for idx in removed_indexes if idx not in remove_from_removed
+            ]
+
+            # Find index_together changed to indexes.
+            remove_from_added = []
+            for (
+                old_value,
+                new_value,
+                index_together_app_label,
+                index_together_model_name,
+                dependencies,
+            ) in self._get_altered_foo_together_operations(
+                operations.AlterIndexTogether.option_name
+            ):
+                disappearing_values = old_value.difference(new_value)
+                for removed_index_together in disappearing_values:
+                    renamed_index_together_indexes = [
+                        idx
+                        for idx in added_indexes
+                        if idx.fields == list(removed_index_together)
+                        and app_label == index_together_app_label
+                        and model_name == index_together_model_name
```

Comment:
Checking list of fields is not enough. Indexes with any custom options (`condition`, `deferrable`, `include`, `expressions`, etc.) should not be taken into account.